staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -121,6 +122,20 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	case <-after:
 		postTimeoutFn()
 		tw.timeout(err)
+
+		// resultCh needs to have a reader, since the function doing
+		// the work needs to send to it.
+		go func() {
```

Comment:
I think we should move this to be beginning of the "case <-after" clause to ensure that it is invoked. If your concerned  about invoking this prior to the postTimeoutFn() we could always defer it.

Reply:
Fixed.